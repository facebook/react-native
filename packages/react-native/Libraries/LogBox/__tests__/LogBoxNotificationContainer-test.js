/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../jest/renderer');
const LogBoxLog = require('../Data/LogBoxLog').default;
const {
  _LogBoxInspectorContainer: LogBoxInspectorContainer,
} = require('../LogBoxInspectorContainer');
const React = require('react');

// Mock `LogBoxInspector` because we are interested in snapshotting the behavior
// of `LogBoxNotificationContainer`, not `LogBoxInspector`.
jest.mock('../UI/LogBoxInspector', () => ({
  __esModule: true,
  default: 'LogBoxInspector',
}));

describe('LogBoxNotificationContainer', () => {
  it('should render inspector with logs, even when disabled', async () => {
    const output = await render.create(
      <LogBoxInspectorContainer
        isDisabled
        selectedLogIndex={-1}
        logs={[
          new LogBoxLog({
            level: 'warn',
            isComponentError: false,
            message: {
              content: 'Some kind of message',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message',
            componentStack: [],
          }),
          new LogBoxLog({
            level: 'error',
            isComponentError: false,
            message: {
              content: 'Some kind of message (latest)',
              substitutions: [],
            },
            stack: [],
            category: 'Some kind of message (latest)',
            componentStack: [],
          }),
        ]}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
