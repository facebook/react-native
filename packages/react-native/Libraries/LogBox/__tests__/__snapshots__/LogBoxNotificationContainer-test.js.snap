// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxNotificationContainer should render inspector with logs, even when disabled 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <LogBoxInspector
    logs={
      Array [
        LogBoxLog {
          "category": "Some kind of message",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "warn",
          "message": Object {
            "content": "Some kind of message",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        },
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "componentStackType": "legacy",
          "count": 1,
          "extraData": undefined,
          "isComponentError": false,
          "level": "error",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "onNotificationPress": undefined,
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
          "symbolicatedComponentStack": Object {
            "componentStack": null,
            "error": null,
            "status": "NONE",
          },
          "type": undefined,
        },
      ]
    }
    onChangeSelectedIndex={[Function]}
    onDismiss={[Function]}
    onMinimize={[Function]}
    selectedIndex={-1}
  />
</View>
`;
